subsys/bluetooth/mesh/lpn.c

```diff
@@ -38,6 +38,12 @@ LOG_MODULE_REGISTER(bt_mesh_lpn);
 	 CONFIG_LOG_MODE_DEFERRED Kconfig option when Low Power node feature is enabled.
 #endif
 
+#if defined(CONFIG_BT_MESH_ADV_LEGACY)
+#define RX_DELAY_CORRECTION ((int32_t)lpn->adv_duration)
```

Comment:
`lpn` should be a macro parameter here. Macros shouldn't make assumptions of the environment in which they're called (e.g. the availability or naming of specific variables in the current scope).